nca/FWRules/ConnectivityGraph.py

```diff
@@ -298,39 +298,31 @@ def _get_equals_groups(self):
+    def get_connections_without_fw_rules_txt_format(self, connectivity_msg=None, exclude_self_loop_conns=True):
         """
         :param Union[str,None] connectivity_msg: a msg header describing either the type of connectivity (TCP/non-TCP)
             for connectivity-map output with connectivity restriction,
             or the type of connectivity changes for semantic-diff query output
+        :param bool exclude_self_loop_conns: indicates if to exclude/ include connections from workload to itself
+        - always true for connectivity-map query output
         :rtype: str
         :return: a string of the original peers connectivity graph content (without minimization of fw-rules)
         """
         lines = set()
-        workload_name_to_peers_map = {}  # a dict from workload_name to pods set, to track replicas and copies
         for connections, peer_pairs in self.connections_to_peers.items():
             if not connections:
                 continue
             for src_peer, dst_peer in peer_pairs:
-                src_peer_name = self._get_peer_details(src_peer, True)[0]
-                if src_peer == dst_peer:  # relevant with all connections only
-                    # add the pod to the map with its workload name
-                    if src_peer_name not in workload_name_to_peers_map:
-                        workload_name_to_peers_map[src_peer_name] = {src_peer}
-                    else:
-                        workload_name_to_peers_map[src_peer_name].add(src_peer)
-                    continue  # after having the full dict, lines from pod to itself will be added for workload names
-                    # with only one pod.
-                    # if a peer has different replicas or copies, a connection from it to itself will be added automatically
-                    # only if there are connections between the replicas too (not only from a single pod to itself)
-                dst_peer_name = self._get_peer_details(dst_peer, True)[0]
-                conn_str = connections.get_simplified_connections_representation(True)
-                conn_str = conn_str.title() if not conn_str.isupper() else conn_str
-                lines.add(f'{src_peer_name} => {dst_peer_name} : {conn_str}')
-
-        # adding conns to itself for workloads with single replica
-        for workload_name in [wl for wl in workload_name_to_peers_map if len(workload_name_to_peers_map[wl]) == 1]:
-            lines.add(f'{workload_name} => {workload_name} : All Connections')
+                if src_peer != dst_peer:
+                    src_peer_name = self._get_peer_details(src_peer, True)[0]
+                    dst_peer_name = self._get_peer_details(dst_peer, True)[0]
+                    # no self-loops: if a peer has different replicas or copies, a connection from it to itself will
+                    # not be added either
+                    if exclude_self_loop_conns and src_peer_name == dst_peer_name:
+                        continue
```

Comment:
can you also add a test for semantic-diff, for which a connection between workload to itself is in the output of the connectivity diff in this format?

Reply:
ok, great! how did it not fail in one of the previous commits on which such connections were removed from the output? 
Is that example tested on txt_no_fw_rules format?  if not, let's add this format to that test as well.

Reply:
now I noticed you've added this format to that test.. thanks!